### ISSUE TITLE
Updating games stuck at 0% 

### ISSUE THREAD

**Issue by Benisgo** [Heroic-Games-Launcher/HeroicGamesLauncher#2642]
### Describe the bug

Updating games using heroic doesn't tell you the update size, How much time is left or how much you've downloaded
The graph at the top for download speed doesn't work either even though in task manager Heroic is downloading updates (Even though I dont know how much it's gonna download)

### Add logs

```text
(02:58:17) INFO:    [Connection]:       Connectivity: check-online
(02:58:17) INFO:    [Connection]:       Pinging external endpoints
(02:58:17) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version
(02:58:19) INFO:    [Legendary]:        Legendary location: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary
(02:58:19) INFO:    [Gog]:              GOGDL location: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl
(02:58:19) INFO:    [Gog]:              GOGDL location: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl
(02:58:19) WARNING: [Backend]:          Protocol already registered.
(02:58:19) INFO:    [Connection]:       Connectivity: online
(02:58:19) INFO:    [Backend]:          User Not Found, removing it from Store
(02:58:19) INFO:    [Gog]:              Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\berry\AppData\Roaming\heroic\gog_store\auth.json --version
(02:58:20) INFO:    [Backend]:          VCRuntime is installed
(02:58:20) INFO:    [Backend]:          Loading Screen Ready
(02:58:20) INFO:    [Backend]:          Frontend Ready
(02:58:21) INFO:    [Frontend]:         Updated to a new version, cleaaning up the cache.
(02:58:21) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary cleanup
(02:58:22) WARNING: [Backend]:          Could not determine System Info
(02:58:25) INFO:    [Backend]:          Starting the Download Queue
(02:58:42) INFO:    [Legendary]:        Logging in: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary auth --code <redacted>
(02:58:44) INFO:    [Frontend]:         Refreshing Library
(02:58:44) INFO:    [Legendary]:        Refreshing library...
(02:58:44) INFO:    [Legendary]:        Refreshing Epic Games...
(02:58:44) INFO:    [Legendary]:        Game list updated, got 295 games & DLCs
(02:58:44) INFO:    [Frontend]:         Called Login
(02:58:44) INFO:    [Frontend]:         No cache found, getting data from legendary...
(02:58:44) INFO:    [Legendary]:        Refreshing library...
(02:58:44) INFO:    [Legendary]:        Game list updated, got 295 games & DLCs
(02:58:44) INFO:    [Frontend]:         Force Update
(02:58:45) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(02:58:50) INFO:    [ExtraGameInfo]:    Getting ExtraGameInfo data for Dead by Daylight
(02:58:50) INFO:    [ExtraGameInfo]:    Getting PCGamingWiki data for Dead by Daylight
(02:58:50) INFO:    [ExtraGameInfo]:    Getting HowLongToBeat data for Dead by Daylight
(02:58:50) INFO:    [ExtraGameInfo]:    Getting GamesDB data for Dead by Daylight
(02:58:50) INFO:    [Legendary]:        Getting more details with 'legendary info'
(02:58:50) INFO:    [ExtraGameInfo]:    Getting ExtraGameInfo data for Dead by Daylight
(02:58:50) INFO:    [ExtraGameInfo]:    Getting PCGamingWiki data for Dead by Daylight
(02:58:50) INFO:    [ExtraGameInfo]:    Getting HowLongToBeat data for Dead by Daylight
(02:58:50) INFO:    [ExtraGameInfo]:    Getting GamesDB data for Dead by Daylight
(02:58:50) INFO:    [Legendary]:        Getting more details with 'legendary info'
(02:58:50) INFO:    [Legendary]:        Getting more details with 'legendary info'
(02:58:50) INFO:    [Legendary]:        Getting more details with 'legendary info'
(02:58:51) INFO:    [Legendary]:        Getting more details with 'legendary info'
(02:58:51) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info Brill --platform Windows --json
(02:58:51) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info Brill --platform Windows --json
(02:58:51) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info Brill --platform Windows --json
(02:58:51) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info Brill --platform Windows --json
(02:58:52) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info Brill --platform Windows --json
(02:58:52) INFO:    [Legendary]:        Getting Info from Epic API
(02:58:53) DEBUG:   [Backend]:          {"free":123067125760,"diskSize":489433329664,"message":"114.62 GiB / 455.82 GiB","validPath":true}
(02:58:53) INFO:    [Legendary]:        Getting Info from Epic API
(02:59:00) DEBUG:   [Legendary]:        Using cached install info
(02:59:02) DEBUG:   [Backend]:          {"free":123142082560,"diskSize":489433329664,"message":"114.69 GiB / 455.82 GiB","validPath":true}
(02:59:02) DEBUG:   [Backend]:          {"free":123142082560,"diskSize":489433329664,"message":"114.69 GiB / 455.82 GiB","validPath":true}
(02:59:10) INFO:    [Legendary]:        Importing Brill.
(02:59:10) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary import --with-dlcs --platform Windows Brill D:\Games\DeadByDaylight
(02:59:10) INFO:    [Backend]:          Preventing machine to sleep
(02:59:10) DEBUG:   [Legendary]:        Using cached install info
(02:59:10) DEBUG:   [Legendary]:        Using cached install info
(02:59:11) ERROR:   [ExtraGameInfo]:    Was not able to get PCGamingWiki data for Dead by Daylight Error: connect ETIMEDOUT 172.67.129.174:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16)
(02:59:11) ERROR:   [ExtraGameInfo]:    Was not able to get PCGamingWiki data for Dead by Daylight Error: connect ETIMEDOUT 172.67.129.174:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16)
(02:59:13) INFO:    [Legendary]:        installed.json updated, refreshing library
(02:59:13) INFO:    [Legendary]:        installed.json updated, refreshing library
(02:59:13) INFO:    [Backend]:          Adding shortcuts for Dead by Daylight
(02:59:13) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Dead by Daylight
(02:59:13) INFO:    [Backend]:          imported Dead by Daylight
(02:59:14) INFO:    [Shortcuts]:        Prepare Steam images for Dead by Daylight
(02:59:16) INFO:    [Shortcuts]:        Prepare Steam images for Dead by Daylight
(02:59:18) INFO:    [Shortcuts]:        Dead by Daylight was successfully added to Steam.
(02:59:18) INFO:    [Frontend]:         Refreshing Library
(02:59:18) INFO:    [Legendary]:        Refreshing library...
(02:59:18) INFO:    [Legendary]:        Game list updated, got 296 games & DLCs
(02:59:18) DEBUG:   [Legendary]:        Using cached install info
(02:59:18) INFO:    [Frontend]:         Force Update
(02:59:25) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Dead by Daylight
(02:59:25) DEBUG:   [Legendary]:        Using cached install info
(02:59:25) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Dead by Daylight
(02:59:25) DEBUG:   [Legendary]:        Using cached install info
(02:59:25) DEBUG:   [Legendary]:        Using cached install info
(02:59:28) INFO:    [Backend]:          Launching Dead by Daylight (Brill)
(02:59:28) INFO:    [Backend]:          Preventing display from sleep
(02:59:28) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version
(02:59:28) INFO:    [Backend]:          Preventing machine to sleep
(02:59:28) DEBUG:   [Legendary]:        Using cached install info
(02:59:28) DEBUG:   [Legendary]:        Using cached install info
(02:59:28) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary launch Brill --json --offline
(02:59:28) INFO:    [Backend]:          Stopping Power Saver Blocker
(02:59:28) DEBUG:   [Legendary]:        Using cached install info
(02:59:28) DEBUG:   [Legendary]:        Using cached install info
(02:59:28) INFO:    [Gog]:              Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\berry\AppData\Roaming\heroic\gog_store\auth.json --version
(02:59:29) INFO:    [Legendary]:        Launching Dead by Daylight: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary launch Brill --language en
(02:59:31) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(02:59:31) INFO:    [Frontend]:         Refreshing Library
(02:59:31) INFO:    [Legendary]:        Refreshing library...
(02:59:31) INFO:    [Legendary]:        Game list updated, got 296 games & DLCs
(02:59:31) DEBUG:   [Legendary]:        Using cached install info
(02:59:31) DEBUG:   [Legendary]:        Using cached install info
(02:59:33) WARNING: [Backend]:          Could not determine System Info
(03:01:27) DEBUG:   [Legendary]:        Using cached install info
(03:01:27) INFO:    [DownloadManager]:  Dead by Daylight  was added to the download queue.
(03:01:27) DEBUG:   [Legendary]:        Using cached install info
(03:01:27) DEBUG:   [Legendary]:        Using cached install info
(03:01:28) DEBUG:   [Legendary]:        Using cached install info
(03:01:28) INFO:    [Legendary]:        Updating Brill: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary update Brill -y
(03:01:28) DEBUG:   [Legendary]:        Logging to file "C:\Users\berry\AppData\Roaming\heroic\GamesConfig\Brill.log"
(03:01:28) DEBUG:   [Legendary]:        Using cached install info
(03:01:28) DEBUG:   [Legendary]:        Using cached install info
(03:21:32) INFO:    [Backend]:          Trying to kill legendary
(03:21:33) INFO:    [Backend]:          Killed legendary
(03:21:33) INFO:    [Legendary]:        Abort command "C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary update Brill -y"
(03:21:33) INFO:    [DownloadManager]:  Finished Update of Brill
(03:21:33) INFO:    [Frontend]:         Refreshing Library
(03:21:33) INFO:    [Legendary]:        Refreshing library...
(03:21:33) INFO:    [Legendary]:        Refreshing Epic Games...
(03:21:33) DEBUG:   [Legendary]:        Update is available for 602eb4abc8764c87b7f2607a1ef8c18e: 4878570-30153 != 4352238-29145
(03:21:33) DEBUG:   [Legendary]:        Update is available for Brill: 5.5.2.3 != DBD_Live_EGS_Shipping_159_1180623_6.7.0.2
(03:21:33) INFO:    [Legendary]:        Found 2 games to update
(03:21:33) INFO:    [Legendary]:        Game list updated, got 296 games & DLCs
(03:21:33) INFO:    [Gog]:              Found 0 game(s) to update
(03:21:33) INFO:    [Frontend]:         Refreshing Library
(03:21:33) INFO:    [Legendary]:        Refreshing library...
(03:21:33) INFO:    [Legendary]:        Refreshing Epic Games...
(03:21:33) DEBUG:   [Legendary]:        Update is available for 602eb4abc8764c87b7f2607a1ef8c18e: 4878570-30153 != 4352238-29145
(03:21:33) DEBUG:   [Legendary]:        Update is available for Brill: 5.5.2.3 != DBD_Live_EGS_Shipping_159_1180623_6.7.0.2
(03:21:33) INFO:    [Legendary]:        Found 2 games to update
(03:21:33) INFO:    [Legendary]:        Game list updated, got 296 games & DLCs
(03:21:33) INFO:    [Gog]:              Found 0 game(s) to update
(03:21:33) INFO:    [Backend]:          Stopping Power Saver Blocker
(03:21:34) INFO:    [Backend]:          operation interrupted
(03:21:34) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(03:21:34) INFO:    [Legendary]:        Running command: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(03:21:38) DEBUG:   [Legendary]:        Using cached install info
(03:21:38) INFO:    [Legendary]:        Updating Brill: C:\Users\berry\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary update Brill -y
(03:21:38) DEBUG:   [Legendary]:        Logging to file "C:\Users\berry\AppData\Roaming\heroic\GamesConfig\Brill.log"
```

### Steps to reproduce

1. Try to update any game using the launcher

### Expected behavior

The download speed graph to be working and for the progress bar to actually function and show how much is left to download of the update

### Screenshots

![image](https://user-images.githubusercontent.com/100228124/233755553-70270f5d-b93c-44b3-bbb0-89e5c33c0607.png)

### Heroic Version

Latest Stable

### System Information

Windows 11

**Comment by flavioislima** [comment]
Cannot replicate. Just updated Cyberpunk2077 and worked just fine.

Can you check the logs under:
C:\Users\berry\AppData\Roaming\heroic\GamesConfig\Brill.log ?

**Comment by arielj** [comment]
Closing since we cannot reproduce this and there's no answer since april. Can be reopened if the issue reappears.

**Comment by xA98Kx** [comment]
![20240104_004800.jpg](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/116504369/f2944e6b-def6-4253-b41d-0b41c770450b)

Happening with me on Ubuntu 23.10

**Comment by BestRazer** [comment]
Same here on Arch
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/65088083/b315d6d3-b408-4e16-9133-c723f909a4cb)

**Comment by JimmJam** [comment]
Same here on NixOS.
![swappy-20240501_172700](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/34753734/7e298f13-729d-4a36-8b92-08328c6a1226)

**Comment by JimmJam** [comment]
Appending this: updates typically do not start for me in general. This is the case for Cyberpunk and Hitman 3, however upon reinstalling Hitman 3 it allowed me to update Cyberpunk. If I encounter this again, I will attempt to start downloading a different game and seeing if that also unlocks the download process. Perhaps updating does not trigger the required connection to Epic's server in the same way downloading initiates hence updates working from then on?

**Comment by JimmJam** [comment]
I have just checked the logs when updating GTA 5, and it appears legendary does a LOT of verifying before it updates. Like, 100GB worth of verifying the launcher doesn't give any notice about, might be causing this misunderstanding..

**Comment by Wrobel1111** [comment]
I had a similar case with updating Dead by Daylight with Heroic 2.17.2
The comment from @JimmJam made me check the logs for Epic/Legendary - his observation was correct. Maybe it is worth somehow showing the process of verification? Especially that the speed of verification is present in the logs, but the monitor in the "Downloading" tab shows 0 MB/s
This comment might be a bit of a necromancy, but I'm still bumping this up, since the issue is still present

**Comment by viceebun** [comment]
I have still encountered this issue from time to time, an apparent lack of any attempt made from the backend Legendary to relay its verification through Heroic, leaving it to display 0% for an ungodly amount of time. I second the reopening/bump

**Comment by givname** [comment]
hitting the same issue waiting over an hour after finishing download, still not complete verifying
<img width="1609" height="1013" alt="Image" src="https://github.com/user-attachments/assets/ab466ec3-6908-4ea1-8540-7d580d985479" />
meanwhile showing very low bandwith of heroic
<img width="732" height="23" alt="Image" src="https://github.com/user-attachments/assets/db07a6ea-88d4-4cd2-a1b4-25e438086a99" />

**Comment by sdwvit** [comment]
Left an update overnight, it just self cancelled an update. Stalker 2 Zone Kit from Epic store.

```
(08:59:10) [INFO]:    [Legendary]:       Checking for game updates: LEGENDARY_CONFIG_PATH=/home/sdwvit/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/x64/linux/legendary list --third-party
(08:59:10) [DEBUG]:   [Legendary]:       Logging to file(s) "/home/sdwvit/.var/app/com.heroicgameslauncher.hgl/.local/state/Heroic/logs/runners/legendary.log"
(08:59:12) [DEBUG]:   [Legendary]:       Update is available for 90ab4c3f02ce451f9d01dd7be288eb67: 1.6.297 != 1.6.380
(08:59:12) [INFO]:    [Legendary]:       Found 1 game to update
(09:01:21) [DEBUG]:   [Legendary]:       Using cached install info
(09:01:21) [INFO]:    [DownloadManager]: S.T.A.L.K.E.R. 2 Zone Kit  was added to the download queue.
(09:01:21) [DEBUG]:   [Legendary]:       Using cached install info
(09:01:21) [INFO]:    [Legendary]:       Updating 90ab4c3f02ce451f9d01dd7be288eb67: LEGENDARY_CONFIG_PATH=/home/sdwvit/.var/app/com.heroicgameslauncher.hgl/config/heroic/legendaryConfig/legendary /app/bin/heroic/resources/app.asar.unpacked/build/bin/x64/linux/legendary update 90ab4c3f02ce451f9d01dd7be288eb67 -y --skip-sdl
(09:01:21) [DEBUG]:   [Legendary]:       Logging to file(s) "/home/sdwvit/.var/app/com.heroicgameslauncher.hgl/.local/state/Heroic/logs/games/90ab4c3f02ce451f9d01dd7be288eb67_legendary/update.log", "/home/sdwvit/.var/app/com.heroicgameslauncher.hgl/.local/state/Heroic/logs/runners/legendary.log"
(09:01:21) [INFO]:    [Backend]:         Preventing machine to sleep
```

<img height="513" alt="Image" src="https://github.com/user-attachments/assets/3510328a-8ec7-485c-ba36-a88567ee3d07" />

---

#### Update:
did `cat /home/sdwvit/.var/app/com.heroicgameslauncher.hgl/.local/state/Heroic/logs/games/90ab4c3f02ce451f9d01dd7be288eb67_legendary/update.log`

and latest log entries say it's doing verification.

```
=> Verifying large file "Stalker2/Content/Paks/Stalker2-WindowsModEditor.pak": 78% (364651.0/467511.7 MiB) [577.0 MiB/s]
```

Would be good to show a progress bar or at least some UI indication work is in progress.
SDK itself is 700 gb, and at a rate of 500MiB/s it would take roughly 30mins to verify.

---

#### Second Update:

```
Verification progress: 186612/186612 (100.0%) [563.2 MiB/s]
[cli] INFO: Verification finished successfully.
[cli] INFO: Preparing download for "S.T.A.L.K.E.R. 2 Zone Kit" (90ab4c3f02ce451f9d01dd7be288eb67)...
[Core] INFO: Parsing game manifest...
[Core] INFO: Install path: /media/nvme/STALKER2ZoneKit
[Core] INFO: Selected CDN: fastly-download.epicgames.com (https)
[DLM] INFO: Found previously interrupted download. Download will be resumed if possible.
[DLM] INFO: Skipping 186612 files based on resume data.
[cli] INFO: Download size is 0, the game is either already up to date or has not changed. Exiting...
Verification progress: 186612/186612 (100.0%) [563.2 MiB/s]
```

Update self cancelled

### Workaround

Clearing Heroic cache and restarting the update helps and now SDK is getting downloaded

**Comment by carlosrl19** [comment]
Same error on Steam deck, my solution was delete Heroic games launcher cache, then exit program, launch again, cancel update from downloads (don't save downloaded files) and restart update, this solved the problem for me.
